src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryBuilder.cs

```diff
@@ -216,9 +215,23 @@ internal static IQueryMatch BuildQuery(Parameters builderParameters, out OrderMe
                 coraxQuery = ToCoraxQuery(builderParameters, metadata.Query.Where, ref streamingOptimization);
                 coraxQuery = MaterializeWhenNeeded(coraxQuery, ref streamingOptimization);
             }
-            else
+            // just one item, with known field types
+            else if (sortMetadata is [{ FieldType: MatchCompareFieldType.Floating or MatchCompareFieldType.Integer or MatchCompareFieldType.Sequence } sortBy])
```

Comment:
This is the core of the change for [RavenDB-21092](https://issues.hibernatingrhinos.com/issue/RavenDB-21092) Corax streaming optimization - query with just order by

The idea is that if we have an `order by` without a `where`, we just set the scan on the right field.

That is slightly different from the usual `StreamingOptimization` and I'm not sure if this is the right place for this.

Reply:
Looks good